jerry-core/ecma/builtin-objects/ecma-builtin-object.c

```diff
@@ -1151,6 +1156,127 @@ ecma_builtin_object_object_is (ecma_value_t arg1, /**< routine's first argument
 
+/**
+ * The Object object's 'fromEntries' routine
+ *
+ * See also:
+ *          ECMA-262 v10, 19.1.2.7
+ * @return ecma value
+ *         Returned value must be freed with ecma_free_value.
+ */
+static ecma_value_t
+ecma_builtin_object_from_entries (ecma_value_t iterator) /**< object's iterator */
+{
+  /* 1. */
+  if (ECMA_IS_VALUE_ERROR (ecma_op_check_object_coercible (iterator)))
+  {
+    return ECMA_VALUE_ERROR;
+  }
+  /* 2 */
+  ecma_object_t *object_prototype_p = ecma_builtin_get (ECMA_BUILTIN_ID_OBJECT_PROTOTYPE);
+  ecma_object_t *obj_p = ecma_create_object (object_prototype_p, 0, ECMA_OBJECT_TYPE_GENERAL);
+
+  /* 6 */
+  ecma_value_t next_method;
+  ecma_value_t result = ecma_op_get_iterator (iterator, ECMA_VALUE_SYNC_ITERATOR, &next_method);
+
+  if (ECMA_IS_VALUE_ERROR (result))
+  {
+    ecma_free_value (next_method);
+    ecma_deref_object (obj_p);
+    return result;
+  }
+
+  const ecma_value_t temp_iterator = result;
+
+  while (true)
+  {
+    result = ecma_op_iterator_step (temp_iterator, next_method);
+
+    if (ECMA_IS_VALUE_ERROR (result))
+    {
+      goto cleanup_iterator;
+    }
+
+    if (ecma_is_value_false (result))
+    {
+      break;
+    }
+
+    const ecma_value_t next = result;
+    result = ecma_op_iterator_value (next);
+    ecma_free_value (next);
+
+    if (ECMA_IS_VALUE_ERROR (result))
+    {
+      goto cleanup_iterator;
+    }
+
+    if (!ecma_is_value_object (result))
+    {
+      ecma_free_value (result);
+      ecma_raise_type_error (ECMA_ERR_MSG ("Iterator value is not an object."));
+      result = ecma_op_iterator_close (temp_iterator);
+      JERRY_ASSERT (ECMA_IS_VALUE_ERROR (result));
+      goto cleanup_iterator;
+    }
+
+    ecma_object_t *next_object_p = ecma_get_object_from_value (result);
+
+    result = ecma_op_object_get_by_index (next_object_p, 0);
+
+    if (ECMA_IS_VALUE_ERROR (result))
+    {
+      ecma_deref_object (next_object_p);
+      ecma_op_iterator_close (temp_iterator);
+      goto cleanup_iterator;
+    }
+
+    const ecma_value_t key = result;
+
+    result = ecma_op_object_get_by_index (next_object_p, 1);
+
+    if (ECMA_IS_VALUE_ERROR (result))
+    {
+      ecma_deref_object (next_object_p);
+      ecma_free_value (key);
+      ecma_op_iterator_close (temp_iterator);
+      goto cleanup_iterator;
+    }
+
+    const ecma_value_t value = result;
+    ecma_string_t *property_key = ecma_op_to_property_key (key);
+    ecma_property_value_t *prop;
+
+    prop = ecma_create_named_data_property (obj_p,
+                                            property_key,
+                                            ECMA_PROPERTY_CONFIGURABLE_ENUMERABLE_WRITABLE,
+                                            NULL);
+    prop->value = ecma_copy_value_if_not_object (value);
+
+    ecma_deref_ecma_string (property_key);
+    ecma_free_value (key);
+    ecma_free_value (value);
+    ecma_deref_object (next_object_p);
+
+    if (ECMA_IS_VALUE_ERROR (result))
```

Comment:
result cannot be an abrupt completion here since it's checked in line 1239.

Reply:
sorry i misread